src/oph/ehoks/external/eperusteet.clj

```diff
@@ -55,6 +55,48 @@
          (remove-empty-kriteerit)
          (adjust-osaamistaso-based-on-asteikko asteikko))))
 
+(defn- get-peruste-by-id
+  "Get perusteet by ID. Uses eperusteet external api."
+  [^Long id]
+  (let [result (c/with-api-headers
+                 {:method :get
+                  :service (u/get-url "eperusteet-service-url")
+                  :url (u/get-url "eperusteet-service.external-api.find-peruste"
+                                  id)
+                  :options {:as :json}})
+        body (:body result)]
+    body))
+
+(defn get-koulutuksenOsa-by-koodiUri
+  "Search for perusteet that match a koodiUri. Uses eperusteet external api."
+  [^String koodiUri]
+  (let [result
+        (c/with-api-headers
+          {:method :get
+           :service (u/get-url "eperusteet-service-url")
+           :url (u/get-url
+                  "eperusteet-service.external-api.find-perusteet-by-koodi")
+           :options {:as :json
+                     :query-params {:koodi koodiUri}}})
+        body (get-in result [:body :data])
+        id (:id (first body))
+        peruste (get-peruste-by-id id)
```

Comment:
With this function, aren't you getting exactly the same peruste as with `(first body)` one line above?